aptos-move/e2e-move-tests/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+    pub fn new_account_at(&mut self, addr: AccountAddress) -> Account {
+        // The below will use the genesis keypair but that should be fine.
+        let acc = Account::new_genesis_account(addr);
+        let data = AccountData::with_account(acc, 1_000_000, 10);
+        self.executor.add_account_data(&data);
+        data.account().clone()
+    }
+
+    /// Runs a transaction, based on provided payload. If the transaction succeeds, any generated
+    /// writeset will be applied to storage.
+    pub fn run_txn(&mut self, account: &Account, payload: TransactionPayload) -> TransactionStatus {
+        let seq_no = if let Some(n) = self.txn_seq_no.get(account.address()) {
+            *n
+        } else {
+            10 // that is what we used above to initialize the account with
+        };
+        let txn = account
+            .transaction()
+            .sequence_number(seq_no)
+            .gas_unit_price(1)
+            .payload(payload)
+            .sign();
+        let output = self.executor.execute_transaction(txn);
+        if matches!(output.status(), TransactionStatus::Keep(_)) {
+            self.executor.apply_write_set(output.write_set());
+            self.txn_seq_no.insert(*account.address(), seq_no + 1);
+        }
+        output.status().to_owned()
+    }
+
+    /// Run the specified entry point `fun`. Arguments need to be provided in bcs-serialized form.
+    pub fn run_entry_txn(
+        &mut self,
+        account: &Account,
+        fun: MemberId,
+        ty_args: Vec<TypeTag>,
+        args: Vec<Vec<u8>>,
+    ) -> TransactionStatus {
+        let MemberId {
+            module_id,
+            member_id: function_id,
+        } = fun;
+        self.run_txn(
+            account,
+            TransactionPayload::ScriptFunction(ScriptFunction::new(
+                module_id,
+                function_id,
+                ty_args,
+                args,
+            )),
+        )
+    }
+
+    /// Publishes the Move Package found at the given path on behalf of the given account.
+    pub fn publish_package(
+        &mut self,
+        account: &Account,
+        path: &Path,
+        upgrade_policy: UpgradePolicy,
+    ) -> TransactionStatus {
+        let package = BuiltPackage::build(MovePackageDir::new(path.to_owned()), true, false)
+            .expect("building package must succeed");
+        let code = package.extract_code();
+        let metadata = package
+            .extract_metadata(upgrade_policy)
+            .expect("extracting package metdata must succeed");
+        self.run_txn(
+            account,
+            aptos_stdlib::code_publish_package_txn(
+                bcs::to_bytes(&metadata).expect("PackageMetadata has BCS"),
+                code,
+            ),
+        )
+    }
+
+    /// Reads the raw, serialized data of a resource.
+    pub fn read_resource_raw(&self, addr: &AccountAddress, struct_id: MemberId) -> Option<Vec<u8>> {
```

Comment:
This will likely need to be updated to support TypeParams early on. Any reason to not just support StructTag instead of "MemberId"?

Reply:
`StructTag` does not has FromStr which I find essential for readability, so you can say `parse("0x1::code::Registry").unwrap()`. An actually full blown type parser would be nice but we'd need to start in the Move repo. I added the type args as additional parameter.

Reply:
Ok, I found now `parse_struct_tag` so we are fine with this.